debezium-core/src/main/java/io/debezium/metrics/Metrics.java

```diff
@@ -72,6 +74,7 @@ protected Metrics(CommonConnectorConfig connectorConfig, String contextName, boo
      */
     public synchronized void register() {
         try {
+            final Metronome metronome = Metronome.sleeper(REGISTRATION_RETRY_DELAY, Clock.system());
```

Comment:
Could you please resturcture the code to mimic https://github.com/debezium/debezium/blob/main/debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java#L345 ?

Reply:
Done.